Zend/zend_alloc.c

```diff
@@ -2261,8 +2269,12 @@ void zend_mm_shutdown(zend_mm_heap *heap, bool full, bool silent)
 	zend_mm_chunk *p;
 	zend_mm_huge_list *list;
 
+	if (full == false) {
+		zend_mm_observers_shutdown(heap);
+	}
```

Comment:
Why is this only called on `full == false`? It sounds like this would leak the observers when shutting down, which would be reported by asan. Hard to tell without tests.

Reply:
Found this as well in the morning when playing further with it, I'll incorporate your suggestions and push my latest changes.